Sources/Sentry/SentryClient.m

```diff
@@ -313,6 +314,16 @@ - (void)restoreContextBeforeCrash {
     self.lastContext = context;
 }
 
+- (void)setSampleRate:(float)sampleRate {
+    if (sampleRate < 0 && sampleRate > 1) {
+        return;
+    }
+    _sampleRate = sampleRate;
+    self.shouldSendEvent = ^BOOL(SentryEvent *_Nonnull event) {
+        return (sampleRate >= ((double)arc4random() / 0x100000000));
```

Comment:
Not that it matters much, but arc4random's upper bound is 0xffffffff

Reply:
Are you sure?
http://nshipster.com/random/
`arc4random has a range up to 0x100000000 (4294967296), whereas rand and random top out at RAND_MAX = 0x7fffffff (2147483647).`